<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ValidationToolbox</class>
 <widget class="QDockWidget" name="ValidationToolbox">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>426</width>
    <height>610</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>&amp;DSGTools: Validation Toolbox</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QGridLayout" name="gridLayout_15">
    <item row="0" column="0">
     <widget class="QTabWidget" name="validationTabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Configure </string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Processes</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0" colspan="2">
         <widget class="ConnectionComboBox" name="connectionSelectorComboBox" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>60</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QgsCollapsibleGroupBox" name="ProcessesGroupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>200</height>
           </size>
          </property>
          <property name="title">
           <string>Validation Processes</string>
          </property>
          <property name="saveCollapsedState">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLineEdit" name="filterLineEdit">
             <property name="placeholderText">
              <string>Filter by process name</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QTreeWidget" name="processTreeWidget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="dragEnabled">
              <bool>false</bool>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::NoDragDrop</enum>
             </property>
             <column>
              <property name="text">
               <string>Available Processes</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox_7">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>Validation Workflow</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="ValidationWorkflowCompactPropertyManagerWidget" name="validationWorkflowPropertyManagerWidget" native="true"/>
           </item>
           <item row="1" column="0">
            <widget class="QTreeWidget" name="workflowTreeWidget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <attribute name="headerCascadingSectionResizes">
              <bool>true</bool>
             </attribute>
             <attribute name="headerMinimumSectionSize">
              <number>1</number>
             </attribute>
             <attribute name="headerStretchLastSection">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string>#</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Task</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Status</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="reRunButton">
          <property name="toolTip">
           <string>Re-Run Last Process</string>
          </property>
          <property name="text">
           <string>Re-Run Workflow</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../resources.qrc">
            <normaloff>:/plugins/DsgTools/icons/re-run.png</normaloff>:/plugins/DsgTools/icons/re-run.png</iconset>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QPushButton" name="runButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Run Process</string>
          </property>
          <property name="text">
           <string>Run Process Workflow</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../resources.qrc">
            <normaloff>:/plugins/DsgTools/icons/run.png</normaloff>:/plugins/DsgTools/icons/run.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Flags</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QComboBox" name="filterTypeComboBox">
          <item>
           <property name="text">
            <string>Process Name</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Class Name</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="DsgCustomComboBox" name="customFilterComboBox" native="true"/>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QTableView" name="tableView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>60</number>
          </attribute>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="clearSelectedPushButton">
          <property name="text">
           <string>Clear Filtered Flags</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../resources.qrc">
            <normaloff>:/plugins/DsgTools/icons/flags.png</normaloff>:/plugins/DsgTools/icons/flags.png</iconset>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="clearAllPushButton">
          <property name="text">
           <string>Clear All Flags</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../resources.qrc">
            <normaloff>:/plugins/DsgTools/icons/flags_all.png</normaloff>:/plugins/DsgTools/icons/flags_all.png</iconset>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="historyButton">
          <property name="text">
           <string>Open History</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../resources.qrc">
            <normaloff>:/plugins/DsgTools/icons/history.png</normaloff>:/plugins/DsgTools/icons/history.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Spatial Rules</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <property name="margin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QRadioButton" name="spatialRuleEnforcerRadio">
             <property name="text">
              <string>Start/Stop Spatial Rule Enforcer</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="GenericCompactPropertyManagerWidget" name="spatialRulePropertyManagerWidget" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Attribute Rules</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="margin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QRadioButton" name="attributeRuleEnforcerRadio">
             <property name="text">
              <string>Start/Stop Attrib&amp;ute Rule Enforcer</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="AttributeRulesCompactPropertyManagerWidget" name="attributeRulePropertyManagerWidget" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Validation Workflows</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6"/>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::MinimumExpanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgis.gui</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DsgCustomComboBox</class>
   <extends>QWidget</extends>
   <header>DsgTools.gui.CustomWidgets.BasicInterfaceWidgets.dsgCustomComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GenericCompactPropertyManagerWidget</class>
   <extends>QWidget</extends>
   <header>DsgTools.gui.CustomWidgets.DatabasePropertiesWidgets.CompactPropertyWidgets.genericCompactPropertyManagerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AttributeRulesCompactPropertyManagerWidget</class>
   <extends>QWidget</extends>
   <header>DsgTools.gui.CustomWidgets.DatabasePropertiesWidgets.CompactPropertyWidgets.attributeRulesCompactPropertyManagerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ConnectionComboBox</class>
   <extends>QWidget</extends>
   <header>DsgTools.gui.CustomWidgets.ConnectionWidgets.AdvancedConnectionWidgets.connectionComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ValidationWorkflowCompactPropertyManagerWidget</class>
   <extends>QWidget</extends>
   <header>DsgTools.gui.CustomWidgets.DatabasePropertiesWidgets.CompactPropertyWidgets.validationWorkflowCompactPropertyManagerWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../resources.qrc"/>
 </resources>
 <connections/>
</ui>
